src/Cool/DidYouKnow/didYouKnowModal.cpp

```diff
@@ -0,0 +1,84 @@
+#include "didYouKnowModal.hpp"
+#include <imgui.h>
+#include <array>
+#include <chrono>
+#include <iostream>
+#include <string>
+#include "fmt/compile.h"
+
+static constexpr const char* const id_did_you_know = "Did you know?";
+
+void Cool::DidYouKnowModal::open()
+{
+    _is_open = true;
+    ImGui::OpenPopup(id_did_you_know);
+    _timestamp_last_opening = std::chrono::steady_clock::now();
+}
+
+void Cool::DidYouKnowModal::open_ifn()
+{
+    auto const difference = std::chrono::steady_clock::now() - _timestamp_last_opening;
+
+    if (!_is_open && difference > timeToWait)
+    {
+        _is_open = true;
+        ImGui::OpenPopup(id_did_you_know);
+        _timestamp_last_opening = std::chrono::steady_clock::now();
+    }
+}
+
+void Cool::DidYouKnowModal::prepare_next_tip()
+{
+    _current_tip_index++;
+    if (_current_tip_index >= all_tips.size())
+        _current_tip_index = 0;
+}
+
+void Cool::DidYouKnowModal::imgui_windows()
+{
+    if (ImGui::BeginPopupModal(id_did_you_know, nullptr, ImGuiWindowFlags_AlwaysAutoResize))
```

Comment:
Find a way to make this popup wrap when the text gets too long.
![image](https://user-images.githubusercontent.com/45451201/231144616-1dbb60c2-8e5d-495e-bbaf-8b24110129e6.png)